keras_core/export/export_lib.py

```diff
@@ -81,32 +83,65 @@ def __init__(self):
         self._endpoint_signatures = {}
         self.tensorflow_version = tf.__version__
 
-        self._tf_trackable = tf.__internal__.tracking.AutoTrackable()
-        self._tf_trackable.variables = []
-        self._tf_trackable.trainable_variables = []
-        self._tf_trackable.non_trainable_variables = []
+        if backend.backend() == "tensorflow":
+            self._tf_trackable = tf.__internal__.tracking.AutoTrackable()
+            self._tf_trackable.variables = []
+            self._tf_trackable.trainable_variables = []
+            self._tf_trackable.non_trainable_variables = []
+        else:  # JAX backend
+            self.module = tf.Module()
+
+            # Variables of `tf.Module` cannot be directly set.
+            self._variables = []
+            self._trainable_variables = []
+            self._non_trainable_variables = []
+        if backend.backend() not in ("tensorflow", "jax"):
+            raise NotImplementedError(
+                "The export API is only compatible with JAX and TF backends."
+            )
 
     @property
     def variables(self):
-        return self._tf_trackable.variables
+        if backend.backend() == "tensorflow":
+            return self._tf_trackable.variables
+        else:  # JAX backend
+            return self._variables
```

Comment:
There's too much divergence between the JAX and TF paths. Either we find a way to unify (use `_tf_trackable`), or we should have separate subclasses for each backend.

Reply:
Good point, I've unified the code paths with _tf_trackable. Thanks!